autonomic-administration-plugin/src/main/java/br/com/autonomiccs/autonomic/administration/plugin/AdministrationAgent.java

```diff
@@ -202,16 +211,16 @@ private void workOnCluster(ClusterVO cluster, ClusterAdministrationHeuristicAlgo
      * method; then it shutdown idle hosts with (if possible)
      * {@link #shutdownIdleHosts(ClusterVO, ClusterAdministrationHeuristicAlgorithm)} method.
      */
-    private void processCluster(ClusterVO cluster, ClusterAdministrationHeuristicAlgorithm administrationAlgorithm) {
+    protected void processCluster(ClusterVO cluster, ClusterAdministrationHeuristicAlgorithm administrationAlgorithm) {
         long clusterId = cluster.getId();
         if (!autonomicClusterManagementService.canProcessCluster(clusterId, administrationAlgorithm)) {
```

Comment:
@alexandrelimassantana , @GabrielBrascher ,
Here it is a good question. I have talked about this with @alexandrelimassantana before. Remember that I told you we did not check if we were or not adding the cluster twice, and because of that before we start processing It we were checking if we could. So, those two lines are needed for these tasks.
We the new component you are introducing, that will enable us to use JADE and create true agents, that can be fixed.